x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/action/RollupShardIndexerStatus.java

```diff
@@ -7,13 +7,51 @@
 
 package org.elasticsearch.xpack.core.rollup.action;
 
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Arrays;
+
 /**
  * Status of the rollup indexer task
  */
-public enum RollupShardIndexerStatus {
-    INITIALIZED,
-    STARTED,
-    FAILED,
-    COMPLETED,
-    CANCELLED
+public enum RollupShardIndexerStatus implements Writeable {
+    INITIALIZED(0),
+    STARTED(1),
+    FAILED(2),
+    COMPLETED(3),
+    CANCELLED(4);
+
+    private final int ordinal;
+
+    RollupShardIndexerStatus(int ordinal) {
+        this.ordinal = ordinal;
+    }
+
+    public static RollupShardIndexerStatus readFromStream(final StreamInput in) throws IOException {
+        int ordinal = in.readVInt();
+        return switch (ordinal) {
+            case 0 -> RollupShardIndexerStatus.INITIALIZED;
+            case 1 -> RollupShardIndexerStatus.STARTED;
+            case 2 -> RollupShardIndexerStatus.FAILED;
+            case 3 -> RollupShardIndexerStatus.COMPLETED;
+            case 4 -> RollupShardIndexerStatus.CANCELLED;
+            default -> throw new IllegalArgumentException(
+                "Unknown ["
+                    + RollupShardIndexerStatus.class.getSimpleName()
+                    + "] value ["
+                    + ordinal
+                    + "]. Expected values ["
+                    + Arrays.toString(RollupShardIndexerStatus.values())
+                    + "]"
+            );
+        };
+    }
+
+    @Override
+    public void writeTo(final StreamOutput out) throws IOException {
+        out.writeVInt(this.ordinal);
```

Comment:
Did we before serialise this enum as string?

Reply:
I will check..in that case I will revert it...

Reply:
This was not a Writable before.